modules/core/src/lib/effect-manager.js

```diff
@@ -36,11 +36,24 @@ export default class EffectManager {
     return effects;
   }
 
+  finalize() {
+    this.cleanup();
+  }
+
   // Private
   setEffects(effects = []) {
+    this.cleanup();
     this.effects = effects;
   }
 
+  cleanup() {
+    if (this.effects) {
+      for (const effect of this.effects) {
+        effect.cleanup();
```

Comment:
It's good practice to always remove the reference to an instance after its resources are released. This way there is no ambiguity about its state from the owner class' perspective.